modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -1103,19 +1109,33 @@ private void replaceCallWithConstant(
         Object receiver
     ) {
         Object[] args = new Object[irInvokeCallMemberNode.getArgumentNodes().size()];
+
+        List<Tuple<Integer,String>> argumentsShouldBeConstants  = new ArrayList<>();
         for (int i = 0; i < irInvokeCallMemberNode.getArgumentNodes().size(); i++) {
             ExpressionNode argNode = irInvokeCallMemberNode.getArgumentNodes().get(i);
             IRDConstant constantDecoration = argNode.getDecoration(IRDConstant.class);
             if (constantDecoration == null) {
-                // TODO find a better string to output
-                throw irInvokeCallMemberNode.getLocation()
-                    .createError(
-                        new IllegalArgumentException(
-                            "all arguments to [" + javaMethod.getName() + "] must be constant but the [" + (i + 1) + "] argument isn't"
-                        )
-                    );
+                // offering the symbol name in error message (CastNode was evolved from ESymbol)
+                String argumentName = argNode instanceof CastNode
```

Comment:
How does this handle the case of `2 + x` as an argument?

Reply:
I can't handle the `2 + x` case yet, so if we need to handle this more complex case, I'm going to write a special method to build the original expression from Node.

Reply:
I see that the argNode has a `location` field with some information about the original expression, but it's missing the **length**.   It would be much simpler to use this `location` field (no need to traverse the argNode data structure)